dtfs-central-api/src/services/portal/deal.service.add-deal-cancelled-activity.test.ts

```diff
@@ -38,32 +42,90 @@ describe('PortalDealService - addGefDealCancelledActivity', () => {
   });
 
   describe(`when dealType is ${DEAL_TYPE.GEF}`, () => {
-    it('should call addPortalActivity', async () => {
-      // Arrange
-      addPortalActivityMock.mockResolvedValue({ matchedCount: 1 });
-
-      jest.spyOn(PortalActivityRepo, 'addPortalActivity').mockImplementation(addPortalActivityMock);
-
-      const mockDeal = deal;
-      mockDeal.dealSnapshot.dealType = DEAL_TYPE.GEF;
+    jest.spyOn(PortalActivityRepo, 'addPortalActivity').mockImplementation(addPortalActivityMock);
+
+    const mockDeal = deal;
+    mockDeal.dealSnapshot.dealType = DEAL_TYPE.GEF;
+
+    describe('when cancellationIsInFuture is false', () => {
+      it('should call addPortalActivity with the correct params', async () => {
+        // Act
+        await PortalDealService.addGefDealCancelledActivity({
+          deal: mockDeal,
+          author,
+          cancellationIsInFuture: false,
+          auditDetails,
+        });
+
+        // Assert
+        expect(addPortalActivityMock).toHaveBeenCalledTimes(1);
+
+        const expectedActivity = {
+          type: PORTAL_ACTIVITY_TYPE.DEAL_CANCELLED,
+          label: PORTAL_ACTIVITY_LABEL.DEAL_CANCELLED,
```

Comment:
Need to look into this, it is confusing - I think have this approach in other places. I might have confused with non-portal activites. Will investgiate and confirm/simplify.

Reply:
It feels like the text to display could just be rendered in gef-ui using a mapper, rather than storing it in the database & effectively saving the same information twice

Reply:
I have removed `type` so we use only `label` now.

I think mapper to render text is overkill, instead we can simply have 1x property saved in the DB that can be consumed.